src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -636,6 +638,34 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly cnSegments = remoteData<CopyNumberSeg[]>({
+        await: () => [
+            this.samples
+        ],
+        invoke: () => fetchCopyNumberSegmentsForSamples(this.samples.result)
+    }, []);
+
+    readonly cnSegmentsByChromosome = remoteData<{ [chromosome: string]: MobxPromise<CopyNumberSeg[]> }>({
+        await: () => [
+            this.genes,
+            this.samples
+        ],
+        invoke: () => {
+            if (this.genes.result) {
+                return Promise.resolve(_.reduce(_.uniq(this.genes.result.map(g => g.chromosome)), (map: { [chromosome: string]: MobxPromise<CopyNumberSeg[]> }, chromosome: string) => {
+                    map[chromosome] = remoteData<CopyNumberSeg[]> ({
```

Comment:
sorry i didn't notice this before, do we have to return a map of remoteDatas?  why not just Promise.all of fetchCopyNumberSegmentsForSamples?

Reply:
Actually, this is for the new endpoint where we fetch CN segments by a single chromosome. If we use `Promise.all`, once this promise (`cnSegmentsByChromosome`) is accessed, wouldn't it start fetching all the data for all chromosomes?

Reply:
are you accessing the data on a per chromosome basis?

Reply:
Yes, each tab is only fetching a single chromosome (except the whole genome tab, which uses a different promise), and also checking if the data for that specific chromosome is pending/complete. In the initial implementation, we were just fetching the whole genome data and loading it all at once into the IGV browser. Now, loading only partial data for genes.